Reinstall Native Dependency step failure

Reinstall Native Dependency step failing due to msal not compatible with node 18.20.x

Error message:

v18.20.1
PS C:\code\tes=package> npm install
npm ERR! code 1
npm ERR! path C:\code\tes=package\node_modules\@azure\msal-node-extensions
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node-gyp rebuild
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@10.0.1
npm ERR! gyp info using node@18.20.1 | win32 | x64
npm ERR! gyp info find Python using Python version 3.9.13 found at "C:\Users\kaghiya\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\python.exe"
npm ERR! gyp http GET https://nodejs.org/download/release/v18.20.1/node-v18.20.1-headers.tar.gz
npm ERR! gyp http 200 https://nodejs.org/download/release/v18.20.1/node-v18.20.1-headers.tar.gz
npm ERR! gyp http GET https://nodejs.org/download/release/v18.20.1/SHASUMS256.txt
npm ERR! gyp http GET https://nodejs.org/download/release/v18.20.1/win-x64/node.lib
npm ERR! gyp http 200 https://nodejs.org/download/release/v18.20.1/SHASUMS256.txt
npm ERR! gyp http 200 https://nodejs.org/download/release/v18.20.1/win-x64/node.lib
npm ERR! gyp info find VS using VS2019 (16.11.34601.136) found at:
npm ERR! gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community"
npm ERR! gyp info find VS run with --verbose for detailed information
npm ERR! gyp info spawn C:\Users\kaghiya\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\python.exe
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args 'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
npm ERR! gyp info spawn args 'binding.gyp',
npm ERR! gyp info spawn args '-f',
npm ERR! gyp info spawn args 'msvs',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args 'C:\\code\\tes=package\\node_modules\\@azure\\msal-node-extensions\\build\\config.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args 'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\addon.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args 'C:\\Users\\kaghiya\\AppData\\Local\\node-gyp\\Cache\\18.20.1\\include\\node\\common.gypi',
npm ERR! gyp info spawn args '-Dlibrary=shared_library',
npm ERR! gyp info spawn args '-Dvisibility=default',
npm ERR! gyp info spawn args '-Dnode_root_dir=C:\\Users\\kaghiya\\AppData\\Local\\node-gyp\\Cache\\18.20.1',
npm ERR! gyp info spawn args '-Dnode_gyp_dir=C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp',
npm ERR! gyp info spawn args '-Dnode_lib_file=C:\\\\Users\\\\kaghiya\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\18.20.1\\\\<(target_arch)\\\\node.lib',
npm ERR! gyp info spawn args '-Dmodule_root_dir=C:\\code\\tes=package\\node_modules\\@azure\\msal-node-extensions',
npm ERR! gyp info spawn args '-Dnode_engine=v8',
npm ERR! gyp info spawn args '--depth=.',
npm ERR! gyp info spawn args '--no-parallel',
npm ERR! gyp info spawn args '--generator-output',
npm ERR! gyp info spawn args 'C:\\code\\tes=package\\node_modules\\@azure\\msal-node-extensions\\build',
npm ERR! gyp info spawn args '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp: binding.gyp not found (cwd: C:\code\tes=package\node_modules\@azure\msal-node-extensions) while trying to load binding.gyp
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:271:18)
npm ERR! gyp ERR! stack at ChildProcess.emit (node:events:517:28)
npm ERR! gyp ERR! stack at ChildProcess._handle.onexit (node:internal/child_process:292:12)
npm ERR! gyp ERR! System Windows_NT 10.0.22631
npm ERR! gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd C:\code\tes=package\node_modules\@azure\msal-node-extensions
npm ERR! gyp ERR! node -v v18.20.1
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok
 
npm ERR! A complete log of this run can be found in: C:\Users\kaghiya\AppData\Local\npm-cache\_logs\2024-04-09T22_28_31_031Z-debug-0.log

--- a/eng/pipelines/templates/stages/platform-matrix.json
+++ b/eng/pipelines/templates/stages/platform-matrix.json
@@ -17,11 +17,7 @@
         "Pool": "env:MACPOOL"
       }
     },
-    "NodeTestVersion": [
-      "18.x",
-      "20.x",
-      "21.2.0"
-    ],
+    "NodeTestVersion": ["18.19.x", "20.x", "21.2.0"],
     "TestType": "node",
     "TestResultsFiles": "**/test-results.xml"
   },
@@ -59,10 +55,7 @@
       },
       "TestType": "node",
       "NodeTestVersion": "18.x",
-      "DependencyVersion": [
-        "max",
-        "min"
-      ],
+      "DependencyVersion": ["max", "min"],
       "TestResultsFiles": "**/test-results.xml"
     }
   ]

--- a/eng/pipelines/templates/stages/platform-matrix.json
+++ b/eng/pipelines/templates/stages/platform-matrix.json
@@ -44,7 +44,7 @@
           "TestResultsFiles": "**/test-results.browser.xml"
         }
       },
-      "NodeTestVersion": "18.x"
+      "NodeTestVersion": "18.19.x"
     },
     {
       "Agent": {
@@ -54,7 +54,7 @@
         }
       },
       "TestType": "node",
-      "NodeTestVersion": "18.x",
+      "NodeTestVersion": "18.19.x",
       "DependencyVersion": ["max", "min"],
       "TestResultsFiles": "**/test-results.xml"
     }

